AccessDeniedException: /var/solr/drupal

# Describe the bug

After release [v1.14.3](https://github.com/amazeeio/lagoon/releases/tag/v1.14.3) and template update in #2566
Solr image is not able anymore to create the core directory in `/var/solr` (removed `fsGroup: 0` in pod `securityContext`)

## To Reproduce

Steps to reproduce the behavior:

1. create a test pod using a Solr image e.g. `uselagoon/solr-7.7-drupal `
`kubectl apply -f solr.yaml`

```yaml

apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: solr-test
  labels:
    app.kubernetes.io/instance: solr
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
  storageClassName: gp2
---
apiVersion: v1
kind: Pod
metadata:
  annotations:
    lagoon.sh/version: 21.3.0
  name: solr-test
spec:
  containers:
  - env:
    - name: CRONJOBS
    envFrom:
    - configMapRef:
        name: lagoon-env
    image: uselagoon/solr-7.7-drupal 
    imagePullPolicy: Always
    livenessProbe:
      failureThreshold: 5
      initialDelaySeconds: 90
      periodSeconds: 10
      successThreshold: 1
      tcpSocket:
        port: 8983
      timeoutSeconds: 3
    name: solr-php-persistent
    ports:
    - containerPort: 8983
      protocol: TCP
    readinessProbe:
      failureThreshold: 3
      initialDelaySeconds: 1
      periodSeconds: 3
      successThreshold: 1
      tcpSocket:
        port: 8983
      timeoutSeconds: 1
    resources:
      requests:
        cpu: 10m
        memory: 10Mi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/solr
      name: solr
  imagePullSecrets:
  - name: lagoon-internal-registry-secret
  priority: 10000
  priorityClassName: lagoon-priority-development
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {} 
  volumes:
  - name: solr
    persistentVolumeClaim:
      claimName: solr-test
```
2. get pod logs `kubectl logs solr-test`
```
2021-04-11 17:59:20.109 ERROR (coreLoadExecutor-9-thread-1) [   x:drupal] o.a.s.c.SolrCore :java.nio.file.AccessDeniedException: /var/solr/drupal                             
Caused by: java.nio.file.AccessDeniedException: /var/solr/drupal                                                                                                              
Caused by: org.apache.solr.common.SolrException: java.nio.file.AccessDeniedException: /var/solr/drupal                                                                        
Caused by: java.lang.IllegalStateException: java.nio.file.AccessDeniedException: /var/solr/drupal                                                                             
Caused by: java.nio.file.AccessDeniedException: /var/solr/drupal
```

## Expected behavior

Solr pod should have permissions to create the core directory in `/var/solr` during bootstrap


## Additional context
Probably related also with rootless workloads work in #2481

--- a/images/kubectl-build-deploy-dind/helmcharts/solr/values.yaml
+++ b/images/kubectl-build-deploy-dind/helmcharts/solr/values.yaml
@@ -17,8 +17,8 @@ imagePullSecrets: []
 nameOverride: ""
 fullnameOverride: ""
 
-podSecurityContext: {}
-  # fsGroup: 2000
+podSecurityContext:
+  fsGroup: 0
 
 securityContext: {}
   # capabilities:

